python/az/aro/azext_aro/custom.py

```diff
@@ -458,6 +474,19 @@ def get_network_resources_from_subnets(cli_ctx, subnets, fail):
         if subnet.get("natGateway", None):
             subnet_resources.add(subnet['natGateway']['id'])
 
+        if oc.network_profile.preconfigured_nsg == 'Enabled':
+            if subnet.get("networkSecurityGroup", None):
+                subnet_resources.add(subnet['networkSecurityGroup']['id'])
+                subnet_with_nsg_preattached = subnet_with_nsg_preattached + 1
+
+    # when preconfiguredNSG feature is Enabled we either have all subnets NSG attached or none.
+    if oc.network_profile.preconfigured_nsg == 'Enabled' and \
+        subnet_with_nsg_preattached != 0 and \
+            subnet_with_nsg_preattached != len(subnets):
+        raise ValidationError("""(ValidationError) preconfiguredNSG feature is enabled but NSG not attached for subnets.
+                                  Please makesure all the subnets have network security group attached and retry.
+                                  If issue persists: raise azure support ticket""")
```

Comment:
this is user error, i don't think we should encourage them to open support request?

Reply:
i think a better error message is to tell them which subnet is missing nsg configured?